### PR TITLE
cleanup and @sd-jwt/present package

### LINKED ISSUE #30: Duplication between core and subpackages
With the split into multiple packages there seems to be quite some duplication between the core and new packages. Some methods are just straight up available in both (e.g. swapClaims) and others have different methods but the logic is duplicated across packages.

Just opening an issue so we can clean this up, as it will probably lead to some inconsistencies if we don't update all code in core to depend on the code in the subpackages.

### PR DESCRIPTION
**Pull request by TimoGlastra**
Cleans up the core package a bit by moving some things to the subpackages as well as removing duplicate code

It also adds a new `@sd-jwt/present` package which can be used to get the required disclosures based on a presentation frame

This way `@sphereon/ssi-types` only has to depend on `@sd-jwt/decode` and `@sphereon/pex` has to depend on `@sd-jwt/core` and `@sd-jwt/present`. While users of this library can still use `@sd-jwt/core`

Fixes #30

### CODE REVIEW

**Comment by berendsliedrecht** [comment]
> `@sd-jwt/present`

What do you think about the name `@sd-jwt/presentation`?

File: packages/core/src/sdJwt/disclosures.ts [hunk 1]
```diff
@@ -80,19 +87,22 @@ export class Disclosure {
             return this as unknown as WithCalculateDigestReturnType<HasherImplementation>
         }
 
-        // Calculate digest
-        const hashResult = hasher(this.encoded)
+        const digestResult = disclosureCalculateDigest(
+            this.asJson(),
+            // TODO: string vs HasherAlgorithm
```
Comment (berendsliedrecht) [inline packages/core/src/sdJwt/disclosures.ts:92]: Could you elaborate in this comment or link to an issue? Not too sure what we should fix here.

File: packages/present/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+# Selective Disclosure JWT (SD-JWT) Draft 06 & Selective Disclosure JWT VC 01
```
Comment (berendsliedrecht) [inline packages/present/README.md:1]: We can just link in the package.json to `../../README.md` right? 

I would like to have individual README's in the future though.

**Review by berendsliedrecht — APPROVED** [review]
Great addition! :)

**Comment by TimoGlastra** [comment]
> > `@sd-jwt/present`
> 
> 
> 
> What do you think about the name `@sd-jwt/presentation`? 

Yes probably more future proof. Will change it. But I think /present is already published now right?

**Comment by berendsliedrecht** [comment]
> > > `@sd-jwt/present`
> > 
> > 
> > 
> > What do you think about the name `@sd-jwt/presentation`? 
> 
> Yes probably more future proof. Will change it. But I think /present is already published now right?

Ah yes. Let's just leave it as-is for now :).

**Pull request merged**